### DISCUSSION TITLE
Bluetooth Classic

### DISCUSSION THREAD

**Discussion by Guustaaf** [micropython/micropython#12170] [Core Development] [answered]
Any idea when bluetooth classic will be available?
Simultaneously with wifi on the RPI Pico.
Thank you kindly for answering.

**Comment by gargar69** [comment]
Thanks so much for all the develpoment work everyone. Like Guustaf I am looking forward to getting bluetooth classic on the Pico, running at the same time as wifi. Linking a legacy, classic bluetooth only, inverter onto the wifi network. Thanks GAR

**Comment by jimmo** [comment, marked as answer]
@Guustaaf @gargar69 There is no-one currently working on this that I'm aware of. I have previously prototyped it, but it's a lot of work to get it all the way to completion.

In some ways it's easier than BLE because you can use the already-implemented profiles from the host stack (e.g. A2DP, RFCOM, HID, etc), but this means that every profile has to have its own custom Python bindings implemented.

It would be great to support it though.

**Comment by mattytrentini** [comment]
FWIW, I am aware of an [implementation for the ESP32](https://github.com/shariltumin/esp32-bluetooth-classic-micropython/) that may be worth looking at for inspiration or even implementation. Would be nice to arrive at a common API.

**Comment by Guustaaf** [comment]
OK, not holding by breath then. Hopefully someone will do it.

Thanks.

**Comment by quantrpeter** [comment]
i need to connect nintendo switch joystick, any hero can make microptyhon support Bluetooth classic please?
